tools/solution_drawer.cpp

```diff
@@ -0,0 +1,147 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <set>
+#include <vector>
+#include <random>
+
+#include <SFML/Graphics.hpp>
+#include <SFML/Window.hpp>
+
+sf::RenderWindow& initialize(double size) {
+  sf::VideoMode vm = sf::VideoMode::getDesktopMode();
+  sf::ContextSettings settings;
+  settings.antialiasingLevel = 0;
+  static sf::RenderWindow window(sf::VideoMode(vm.width / size, vm.height / size), "Vehicle Routing Problem", sf::Style::Titlebar | sf::Style::Close | sf::Style::Resize, settings);
+  window.setPosition(sf::Vector2i(vm.width / 2 - vm.width / (size * 2), vm.height / 2 - vm.height / (size * 2)));
+  window.setKeyRepeatEnabled(1);
+  window.setVerticalSyncEnabled(1);
+  return window;
+}
+
+class SolutionDrawer {
+  static constexpr auto ZOOM_FACTOR = 0.9;
+  static constexpr auto RADIUS      = 3.1;
+public:
+  SolutionDrawer(SolutionDrawer&&) = delete;
+  SolutionDrawer(SolutionDrawer const&) = delete;
+  SolutionDrawer(sf::RenderWindow& window): window {window} {}
+
+  auto solution_parse(std::string const& filename) -> void {
+
+    std::fstream fstream {filename, std::ios_base::in};
+    std::string current_line;
+    std::uint32_t route, vehicle, customer;
```

Comment:
I suggest to use `size_t` everywhere instead since we operate with indices in STD library's containers